src/Uno.UI/Controls/Window.macOS.cs

```diff
@@ -44,8 +44,210 @@ public Window(CGRect contentRect, NSWindowStyle aStyle, NSBackingStore buffering
 			Delegate = new WindowDelegate(this);
 			_inputPane = InputPane.GetForCurrentView();
 			_inputPane.Window = this;
+
+			// Register the types that can be dragging into the window (effectively 'AllowDrop').
+			// This must be done before dragging enters the window or dragging methods won't be invoked.
+			// We register for all possible types and then confirm whether they
+			// are actually supported within the draggingEntered and draggingUpdated methods.
+			// This has a minor side effect in the standard UI (zoom effect) but is considered acceptable.
+			RegisterForDraggedTypes(new string[]
+				{
+					NSPasteboard.NSPasteboardTypeUrl,
+					NSPasteboard.NSPasteboardTypeTIFF,
+					NSPasteboard.NSPasteboardTypePNG,
+					NSPasteboard.NSPasteboardTypeHTML,
+					NSPasteboard.NSPasteboardTypeRTF,
+					NSPasteboard.NSPasteboardTypeRTFD,
+					NSPasteboard.NSPasteboardTypeFileUrl,
+					NSPasteboard.NSPasteboardTypeString
+
+					/* For future use
+					UTType.URL,
+					UTType.Image,
+					UTType.HTML,
+					UTType.RTF,
+					UTType.FileURL,
+					UTType.PlainText,
+					*/
+				});
+		}
+
+		/// <summary>
+		/// Method invoked when a drag operation enters the <see cref="NSWindow"/>.
+		/// </summary>
+		/// <remarks>
+		/// 
+		/// While it is never documented directly, DraggingEntered can be added to NSWindow just like NSView.
+		/// Key docs telling this story include:
+		/// 
+		///  (1) NSWindow documentation does not include most NSView drag/drop methods
+		///      https://developer.apple.com/documentation/appkit/nswindow
+		///  (2) Since NSWindow documentation doesn't reference them, they also aren't included in Xamarin
+		///      https://docs.microsoft.com/en-us/dotnet/api/appkit.nswindow?view=xamarin-mac-sdk-14
+		///  (3) However, the 'draggingEntered' method docs reference 'Window' at the very end
+		///      "Invoked when the mouse pointer enters the destination’s bounds rectangle (if it is a view object)
+		///       or its frame rectangle (if it is a window object). "
+		///      https://developer.apple.com/documentation/appkit/nsdraggingdestination/1416019-draggingentered
+		///  (4) In legacy macOS docs this is a little more explitly stated:
+		///      https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/DragandDrop/Concepts/dragdestination.html#//apple_ref/doc/uid/20000977-BAJBJFBG
+		///  (5) Other macOS drag/drop articles and blogs refer to dragging/dropping directly from Window - although
+		///      give no examples.
```

Comment:
😮 Really nice documentation!

I'm wondering however if this is referenced only in old version of the doc, is it possible that it's deprecated?

Reply:
Its possible, good luck finding documentation stating that though :) macOS documentation is quite poor but draggingEntered, current version, references this for NSWindow: https://developer.apple.com/documentation/appkit/nsdraggingdestination/1416019-draggingentered